### PR TITLE
add updated and removed handler for topologyCtrl

### PR DESCRIPTION
**Pull request by sub-mod**
fixes https://github.com/cockpit-project/cockpit/issues/2737
@stefwalter  kindly review.

### CODE REVIEW

File: pkg/kubernetes/topology.js [hunk 1]
```diff
@@ -57,8 +57,22 @@ define([
                 var all = client.select();
                 client.track(all);
                 $(all).on("changed", digest);
+                $(all).on("updated", entity_updated);
+                $(all).on("removed", entity_removed);
                 digest();
 
+                function entity_updated(ev, entity, key, last) {
+                    if($scope.selected && $scope.selected.kind === last.kind && $scope.selected.key === last.key) {
```
Comment (stefwalter) [inline pkg/kubernetes/topology.js]: I think you only need to do something like:

```
if (last === $scope.selected)
    $scope.selected = entity;
```

Does that work?

File: pkg/kubernetes/topology.js [hunk 2]
```diff
@@ -57,8 +57,22 @@ define([
...
+                        $scope.selected = last;
+                    }
+                }
+
+                function entity_removed(ev, entity, key, last) {
+                    if($scope.selected && last == null) {
```
Comment (stefwalter) [inline pkg/kubernetes/topology.js]: I think this condition might look something like:

```
if ($scope.selected === entity)
    $scope.selected = null;
```

Does that work? Also there is no 'last' argument to the "removed" signal.

**Comment by stefwalter** [comment]
Looks like this is in the right direction. Added a few comments. Could you update the TestDashboard integration test, around the area where we select something in the topology view?

File: pkg/kubernetes/topology.js (hunk 1 above)
Comment (sub-mod) [inline pkg/kubernetes/topology.js]: the revision number would be different

File: pkg/kubernetes/topology.js [hunk 3]
```diff
@@ -57,8 +57,22 @@ define([
                 var all = client.select();
                 client.track(all);
                 $(all).on("changed", digest);
+                $(all).on("updated", entity_updated);
+                $(all).on("removed", entity_removed);
                 digest();
 
+                function entity_updated(ev, entity, key, last) {
+                    if($scope.selected.kind === last.kind && $scope.selected.key === last.key) {
+                        $scope.selected = last;
```
Comment (stefwalter) [inline pkg/kubernetes/topology.js]: Here you're setting the selected to the old entity. I don't think you want to do that.

File: pkg/kubernetes/topology.js [hunk 4]
```diff
@@ -57,8 +57,22 @@ define([
...
+                    }
+                }
+
+                function entity_removed(ev, entity, key, last) {
```
Comment (stefwalter) [inline pkg/kubernetes/topology.js]: The ```"removed"``` event doesn't have a ```last``` argument.

**Comment by stefwalter** [comment]
Merged. Works well.

**Pull request closed without merging**